tools/profileBuilder/model/delete.go

```diff
@@ -0,0 +1,48 @@
+// +build go1.9
+
+// Copyright 2018 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+)
+
+// DeleteChildDirs deletes all child directories of the directory specified by
+// `path`.
+// If it fails to delete a child, it halts execution and returns an err immediately.
+func DeleteChildDirs(dir string) (err error) {
+	var children []os.FileInfo
+
+	children, err = ioutil.ReadDir(dir)
+	if err != nil {
+		return
+	}
+
+	for _, child := range children {
+		if child.IsDir() {
+			childPath := strings.Replace(path.Join(dir, child.Name()), `\`, `/`, -1)
```

Comment:
Why is this necessary?

Reply:
My guess it depends on OS. I would use the filepath package instead.

---
In reply to: [168828461](https://github.com/Azure/azure-sdk-for-go/pull/1097#discussion_r168828461) [](ancestors = 168828461)

Reply:
When you say thins you mean the whole delete thing ? because if some profiles go away, they will stick around if we don't delete them

Reply:
I was referring to fixing up the path (I get that deleting is required to clear out old packages).  I guess when running on Windows we end up with a mix of Windows and Linux dir separators which is unfortunate.  If the filepath package abstracts all of this away then I agree let's use that.

Reply:
Neither `filepath` nor `path` are enough, the separators still don't get normalized without a `strings.Replace` call. https://play.golang.org/p/_dupP3C0ek3